File: mail/mail.go
package mail

import (
	"eFuture/config"
	"log"
	"net/smtp"
	"strings"
)

// Send a mail with `subject` and `content` to `receivers`.
func SendMail(subject string, receivers []string, content string) bool {
	isSent := true
	conf := config.Configuration
	auth := smtp.PlainAuth("", conf.Username, conf.Password, conf.Smtp)
	nickname := "eFuture"
	contentType := "\r\nContent-Type: text/plain; charset=UTF-8\r\n\r\n"
	msg := []byte("To: " + strings.Join(receivers, ",") + "\r\n" +
		"From: " + nickname + "<" + conf.Username + ">\r\n" +
		"Subject: " + subject +
		contentType +
		content,
	)
	err := smtp.SendMail(conf.Smtp+":"+conf.SmtpPort, auth, conf.Username, receivers, msg)
	if err != nil {
		log.Printf("Failed to send mail: %v", err)
		isSent = false
	}
	return isSent
}

// This is a example.
//func main() {
//	SendMail("test email", []string{"liuqi0315@gmail.com"}, "This is a test email, and current time is: "+time.Now().String())
//}

File: mail/mail.py
import smtplib
from email.header import Header
from email.mime.text import MIMEText

from config import host, port, username, password
from log import logger

__charset = 'utf-8'


def send_mail(receiver: str, subject: str, message: str):
    message = MIMEText(message, _charset=__charset)
    message['From'] = Header('Future Email <{}>'.format(username))
    message['To'] = Header('<{}>'.format(receiver))
    message['Subject'] = Header(subject)

    try:
        smtp = smtplib.SMTP(host=host, port=port)
        smtp.starttls()
        smtp.login(username, password)
        smtp.sendmail(username, [receiver], message.as_string())
        logger.info('Successfully send')
    except smtplib.SMTPException as e:
        logger.error('Cannot send email: {0}'.format(e))


if __name__ == '__main__':
    send_mail('liuqi0315@gmail.com', 'Test send mail to gmail', 'Sent from python')
